src/components/PostDetails.tsx

```diff
@@ -1,118 +1,72 @@
+/* eslint-disable import/extensions */
 import React, { useEffect, useState } from 'react';
+import {
+  getCommentPosts,
+  addComment,
+  deleteCom,
+} from '../features/ComentSlice';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { CommentData } from '../types/Comment';
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-  const [visible, setVisible] = useState(false);
-
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
+  const dispatch = useAppDispatch();
+  const { comments, loading, error } = useAppSelector(state => state.comments);
 
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  const [visible, setVisible] = useState(false);
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+  useEffect(() => {
     setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post) {
+      dispatch(getCommentPosts(post.id));
     }
-  };
+  }, [post.id, dispatch]);
```

Comment:
fix this linter error
![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/108538921/58e6853b-523a-4efe-ade0-cee15b435aad)